### PR TITLE
lv 4.51-9

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Switches the upstream to https://salsa.debian.org/debian/lv

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/l/lv.rb [hunk 1]
```diff
@@ -33,6 +32,7 @@ class Lv < Formula
 
   # See https://github.com/Homebrew/homebrew-core/pull/53085.
   # Further issues regarding missing headers reported upstream to nrt@ff.iij4u.or.jp
+  # TODO: send MR to https://salsa.debian.org/debian/lv
```
Comment (AkihiroSuda) [inline Formula/l/lv.rb:35]: ( I can't do this right now due to an issue of their gitlab infra: https://salsa.debian.org/salsa/support/-/issues/424 )

File: Formula/l/lv.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 class Lv < Formula
   desc "Powerful multi-lingual file viewer/grep"
-  homepage "https://web.archive.org/web/20160310122517/www.ff.iij4u.or.jp/~nrt/lv/"
-  url "https://web.archive.org/web/20150915000000/www.ff.iij4u.or.jp/~nrt/freeware/lv451.tar.gz"
-  version "4.51"
-  sha256 "e1cd2e27109fbdbc6d435f2c3a99c8a6ef2898941f5d2f7bacf0c1ad70158bcf"
+  # The upstream homepage was "https://web.archive.org/web/20160310122517/www.ff.iij4u.or.jp/~nrt/lv/"
+  homepage "https://salsa.debian.org/debian/lv"
+  url "https://salsa.debian.org/debian/lv/-/archive/debian/4.51-9/lv-debian-4.51-9.tar.gz"
+  version "4.51-9"
```
Comment (AkihiroSuda) [inline Formula/l/lv.rb:6]: `brew audit --strict lv` passes locally, but the CI is failing with `Stable: version 4.51-9 is redundant with version scanned from URL`

**Review by chenrui333 — APPROVED** [review]

File: Formula/l/lv.rb (hunk 1 above)
Comment (AkihiroSuda) [reply Formula/l/lv.rb]: Submitted https://salsa.debian.org/debian/lv/-/merge_requests/3

File: Formula/l/lv.rb [hunk 3]
```diff
@@ -1,11 +1,11 @@
 class Lv < Formula
   desc "Powerful multi-lingual file viewer/grep"
-  homepage "https://web.archive.org/web/20160310122517/www.ff.iij4u.or.jp/~nrt/lv/"
-  url "https://web.archive.org/web/20150915000000/www.ff.iij4u.or.jp/~nrt/freeware/lv451.tar.gz"
-  version "4.51"
-  sha256 "e1cd2e27109fbdbc6d435f2c3a99c8a6ef2898941f5d2f7bacf0c1ad70158bcf"
+  # The upstream homepage was "https://web.archive.org/web/20160310122517/www.ff.iij4u.or.jp/~nrt/lv/"
+  homepage "https://salsa.debian.org/debian/lv"
```
Comment (AkihiroSuda) [inline Formula/l/lv.rb:4]: Marking as a draft, as I guess Debian patches (https://salsa.debian.org/debian/lv/-/tree/master/debian/patches) should be applied

**Review by stefanb — DISMISSED** [review]

File: Formula/l/lv.rb (hunk 3 above)
Comment (AkihiroSuda) [reply Formula/l/lv.rb:4]: done

File: Formula/l/lv.rb [hunk 4]
```diff
@@ -75,24 +75,3 @@ def install
...
---- a/src/guesslocale.c
-+++ b/src/guesslocale.c
-@@ -24,6 +24,7 @@
-
- #include <stdlib.h>
- #include <string.h>
-+#include <ctype.h>
- #include <locale.h>
- #if defined(HAVE_LANGINFO_CODESET)
- #include <langinfo.h>
```
Comment (AkihiroSuda) [inline Formula/l/lv.rb:98]: These `#include` fixes are included in https://salsa.debian.org/debian/lv/-/blob/debian/4.51-9/debian/patches/fix_implicit_decls.diff?ref_type=tags

File: Formula/l/lv.rb [hunk 5]
```diff
@@ -23,25 +23,25 @@ class Lv < Formula
...
   end
 
   # See https://github.com/Homebrew/homebrew-core/pull/53085.
-  # Further issues regarding missing headers reported upstream to nrt@ff.iij4u.or.jp
+  # Being proposed to Debian: https://salsa.debian.org/debian/lv/-/merge_requests/3
   patch :DATA
 
   def install
-    # Work around for newer Clang
-    ENV.append_to_cflags "-Wno-implicit-int" if DevelopmentTools.clang_build_version >= 1403
```
Comment (AkihiroSuda) [inline Formula/l/lv.rb:40]: This is fixed in Debian via https://salsa.debian.org/debian/lv/-/blob/debian/4.51-9/debian/patches/20051030_%25src%25_~use_off_t.diff?ref_type=tags#L2032

**Review by stefanb — APPROVED** [review]

**Pull request merged**